pkg/gds/certman/certs_test.go

```diff
@@ -205,9 +205,227 @@ func (s *certTestSuite) TestCertManager() {
+	// to REJECTED so that it does not get triggered for reissuance.
+	deltaVASP, err := s.fixtures.GetVASP("delta")
+	require.NoError(err)
+	deltaVASP.VerificationStatus = pb.VerificationState_REJECTED
+	require.NoError(s.db.UpdateVASP(deltaVASP))
+
+	// Capture the number of certificate requests on the delta VASP
+	// before reissuance is triggered.
+	previousReqIds, err := models.GetCertReqIDs(charlieVASP)
+	require.NoError(err)
+	previousNumberOfReqs := len(previousReqIds)
+
+	// Call the certman function at 8 days, which should
+	// reissue the VASP's identity certificate, send the
+	// email with the created pkcs12 password and send
+	// the whisper link, as well as notifying the TRISA
+	// admin that reissuance has started.
+	s.updateVaspIdentityCert(charlieVASP, 8)
+	callTime := time.Now()
+	s.certman.HandleCertificateReissuance()
+
+	v, err := s.db.RetrieveVASP(charlieVASP.Id)
+	require.NoError(err)
+
+	reqIDs, err := models.GetCertReqIDs(v)
+	require.NoError(err)
+	require.Equal(len(reqIDs), previousNumberOfReqs+1)
+
+	// Retrieve the latest certificate request for charlie.
+	certReq, err := s.db.RetrieveCertReq(reqIDs[len(reqIDs)-1])
+	require.NoError(err)
+	require.Equal(certReq.Status, models.CertificateRequestState_READY_TO_SUBMIT)
+
+	// Make sure a new secret was created in the secret manager.
+	sm := s.secret.With(certReq.Id)
+	secret, err := sm.GetLatestVersion(context.Background(), "password")
+	require.NoError(err)
+	require.NotNil(secret)
+
+	// Update the secret manager with the password that will decrypt
+	// the fixture certificate used for testing, overriding the randomly
+	// generated password created by updateVaspIdentityCert.
+	require.NoError(sm.AddSecretVersion(context.Background(), "password", []byte("qDhAwnfMjgDEzzUC")))
+
+	// Call the cert request loop once to submit the certificate request and start it's processing.
+	s.certman.HandleCertificateRequests(certDir)
+	v, err = s.db.RetrieveVASP(charlieVASP.Id)
+	require.NoError(err)
+	require.Equal(pb.VerificationState_ISSUING_CERTIFICATE, v.VerificationStatus)
+
+	// On the second call to the cert request loop the certificate should be downloaded and
+	// attached to the VASP.
+	s.certman.HandleCertificateRequests(certDir)
+	v, err = s.db.RetrieveVASP(charlieVASP.Id)
+	require.NoError(err)
+
+	// Retrieve the newly created certificate and ensure it is valid.
+	idCert := v.IdentityCertificate
+	require.NotNil(idCert)
+	require.Greater(int(idCert.Version), 0)
+	require.NotEmpty(idCert.SerialNumber)
+	require.NotEmpty(idCert.Signature)
+	require.NotEmpty(idCert.SignatureAlgorithm)
+	require.NotEmpty(idCert.PublicKeyAlgorithm)
+	require.NotNil(idCert.Subject)
+	require.NotNil(idCert.Issuer)
+	_, err = time.Parse(time.RFC3339, idCert.NotBefore)
+	require.NoError(err)
+	_, err = time.Parse(time.RFC3339, idCert.NotAfter)
+	require.NoError(err)
+	require.False(idCert.Revoked)
+	require.NotEmpty(idCert.Data)
+	require.NotEmpty(idCert.Chain)
```

Comment:
Some additional test cases here would be to check if the VASP ended up in the right state `VERIFIED` and the new certificate request ended up in the right state `COMPLETED`. Those checks should be pretty easy to add, but I won't make it contingent for merging this PR since you added the deliver certs email check already.